llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -6820,9 +6821,7 @@ void BoUpSLP::buildTree_rec(ArrayRef<Value *> VL, unsigned Depth,
 
   // Special processing for sorted pointers for ScatterVectorize node with
   // constant indeces only.
-  if (AreAllSameInsts && UserTreeIdx.UserTE &&
-      UserTreeIdx.UserTE->State == TreeEntry::ScatterVectorize &&
-      !(S.getOpcode() && allSameBlock(VL))) {
+  if (not AreAllSameBlock && AreScatterAllGEPSameBlock) {
```

Comment:
`not` is not used in the repo. Please use `!`.

Reply:
Thanks, fixed.